src/containers/ipfs/hooks/useGetDiscussion.ts

```diff
@@ -1,103 +1,130 @@
 import { useInfiniteQuery } from '@tanstack/react-query';
-import { useEffect, useState } from 'react';
-import { getToLink } from 'src/utils/search/utils';
+import { useState } from 'react';
+import { getLinks } from 'src/utils/search/utils';
 import BigNumber from 'bignumber.js';
 
-const limit = '10';
+export enum LinkType {
+  to = 'to',
+  from = 'from',
+}
 
-const reduceLinks = (data, cid, time) => {
+const reduceLinks = (data, cid, time, type: LinkType) => {
   return data.reduce((acc, item) => {
-    if (item.from === cid) {
-      return [...acc, { cid: item.to, timestamp: time }];
+    if (item[type] === cid) {
+      return [
+        ...acc,
+        { cid: item[type === LinkType.from ? 'to' : 'from'], timestamp: time },
+      ];
     }
     return [...acc];
   }, []);
 };
 
-const reduceParticleArr = (data, cidFrom: string) => {
+const reduceParticleArr = (data: any, cid: string, type: LinkType) => {
   return data.reduce((acc, item) => {
     const { timestamp } = item;
     if (
       item.tx.body.messages[0]['@type'] === '/cyber.graph.v1beta1.MsgCyberlink'
     ) {
-      const cid = item.tx.body.messages[0].links[0].to;
-      return [...acc, { cid, timestamp }];
+      const cid =
+        item.tx.body.messages[0].links[0][
+          type === LinkType.from ? 'to' : 'from'
+        ];
+      return [...acc, { cid, timestamp, type }];
     }
 
     if (
       item.tx.body.messages[0]['@type'] ===
       '/cosmwasm.wasm.v1.MsgExecuteContract'
     ) {
-      const { links } = item.tx.body.messages[0].msg.cyberlink;
-      const linksReduce = reduceLinks(links, cidFrom, timestamp);
-      return [...acc, ...linksReduce];
+      const links = item.tx.body.messages[0].msg?.cyberlink?.links;
+
+      // if (!links) {
+      //   debugger;
+      // }
+
+      if (links) {
+        const linksReduce = reduceLinks(links, cid, timestamp, type);
+        return [...acc, ...linksReduce];
+      }
     }
     return [...acc];
   }, []);
 };
 
-const getTo = async (
+const request = async (
```

Comment:
1. better to use more semantic aka understandable names like requestLinks..
2. callback is unnecessary in this case, can be rewrites unsing useEffect({}, [response]);